File: 2/2.go
package main

import "fmt"

func main() {
	sum := 0
	f0 := 0
	f1 := 1
	f2 := f0 + f1
	for f2 < 4E6 {
		if f2%2 == 0 {
			sum += f2
		}
		f0 = f1
		f1 = f2
		f2 = f0 + f1
	}
	fmt.Println(sum)
}

File: 2/2.rs
fn main() {
    let mut f1 = 0;
    let mut f2 = 1;
    let mut f3 = f1 + f2;
    let mut sum = 0;
    while f3 < 4000000 {
        if f3%2 == 0 {
            sum += f3;
        }
        f1 = f2;
        f2 = f3;
        f3 = f1 + f2;
    }
    println!("{}", sum);
}
